docs/src/concepts/grid_operators.md

```diff
@@ -0,0 +1,110 @@
+# Grid Operators
+
+The gist of the finite difference relies on replacing derivatives by difference quotients anchored on structured grids. We currently support various finite difference operators for fields defined in Cartesian coordinates. The table below summarizes the most common usage of grid operators, with the grid `g::StructuredGrid` and index `I = @index(Global, Cartesian)` defined and `P = Field(backend, grid, location)` is some field defined on the grid `g`.
+
+| Mathematical Formulation | Code |
+|:-------|:------------|
+| $\frac{\partial}{\partial x} P$ | ` ∂x(P, g, I)` |
+| $\frac{\partial}{\partial y} P$ | ` ∂y(P, g, I)` |
+| $\frac{\partial}{\partial z} P$ | ` ∂z(P, g, I)` |
+| $\nabla P$ | ` divg(P, g, I)` |
+
+## Computing the Divergence of a Vector Field
+
+To illustrate the usage of grid operators, we compute the divergence of an vector field $V$ using the `divg` function. We first allocate memory for required fields.
+
+```julia
+V     = VectorField(backend, grid)
+∇V    = Field(backend, grid, Center())
+# use set! to set up the initial vector field...
+```
+
+The kernel that computes the divergence needs to have the grid information passed as for other finite difference operators.
+
+```julia
+@kernel inbounds = true function update_∇!(V, ∇V, g::StructuredGrid, O)
+    I = @index(Global, NTuple)
+    I = I + O
+    ∇V[I...] = divg(V, g, I...)
+end
+```
+
+The kernel can then be launched when required as we detailed in section [Kernels](./kernels.md).
+
+```julia
+launch(arch, grid, update_∇! => (V, ∇V, grid))
+```
+
+
+
+## Masking
```

Comment:
Here I am unsure whether one should make some other example, MWE like thing are or could be used as in FastIce? Maybe @utkinis has some thoughts.